utils/tests/test_hosts.py

```diff
@@ -0,0 +1,162 @@
+        assert self.TestUser1.host is None
+        assert self.TestUser2.host is None
+
+        self.TestUser1.set_host_name()
+        assert self.TestUser1.host == "https://api.exp.move.mil"
+        assert self.TestUser2.host == "https://api.exp.move.mil"
+
+    @mock.patch.dict(os.environ, {"MOVE_MIL_EXP_TLS_CERT": "test_cert", "MOVE_MIL_EXP_TLS_KEY": "test_key"})
+    def test_set_cert_kwargs(self):
+        # Test if the environment is experimental
+        self.HostLocalClass.cert_kwargs = None
+        self.TestUser1.set_cert_kwargs()
+
+        assert re.search("tls/exp_tls_cert_key.pem", self.TestUser1.cert_kwargs["cert"])
+        assert re.search("tls/dod-ca-60-61-bundle.pem", self.TestUser1.cert_kwargs["verify"])
+        assert re.search("tls/exp_tls_cert_key.pem", self.TestUser2.cert_kwargs["cert"])
+        assert re.search("tls/dod-ca-60-61-bundle.pem", self.TestUser2.cert_kwargs["verify"])
+
+        # Test if the environment is local
+        self.HostLocalClass.cert_kwargs = None
+        self.HostLocalClass.env = MilMoveEnv.LOCAL
+        self.TestUser1.set_cert_kwargs()
+
+        assert re.search("tls/devlocal-mtls.cer", self.TestUser1.cert_kwargs["cert"][0])
+        assert re.search("tls/devlocal-mtls.key", self.TestUser1.cert_kwargs["cert"][1])
+        assert re.search("tls/devlocal-mtls.cer", self.TestUser2.cert_kwargs["cert"][0])
+        assert re.search("tls/devlocal-mtls.key", self.TestUser2.cert_kwargs["cert"][1])
+
+    @mock.patch.dict(os.environ, {"MOVE_MIL_EXP_TLS_CERT": "test_cert", "MOVE_MIL_EXP_TLS_KEY": "test_key"})
+    def test_create_deokoyed_cert_file(self):
+        self.HostLocalClass.env = MilMoveEnv.EXP
+        cert_file_path = self.TestUser1.create_deployed_cert_file()
+        test_file_contents = "test_cert\ntest_key"
+
+        assert re.search("tls/exp_tls_cert_key.pem", cert_file_path)
+        with open(cert_file_path, "r") as f:
+            assert f.read() == test_file_contents
+
+    def test_no_env_variables_create_deokoyed_cert_file(self):
+        self.HostLocalClass.env = MilMoveEnv.EXP
+        with pytest.raises(ImplementationError):
+            self.TestUser1.create_deployed_cert_file()
+
+    @mock.patch.dict(os.environ, {"MOVE_MIL_EXP_TLS_CERT": "test_cert", "MOVE_MIL_EXP_TLS_KEY": "test_key"})
+    def test_remove_deployed_cert_file(self):
+        # Setup kwargs
+        self.HostLocalClass.env = MilMoveEnv.EXP
+        self.HostLocalClass.cert_kwargs = None
+        self.TestUser1.set_cert_kwargs()
+        cert_kwargs_before = self.TestUser1.cert_kwargs["cert"]
+
+        self.TestUser1.remove_deployed_cert_file()
+        assert os.path.exists(cert_kwargs_before) is False
+        assert self.TestUser1.cert_kwargs == {}
+
+        # Call function again to make sure it doesn't error
+        self.TestUser1.remove_deployed_cert_file()
+
+
+# tasks.base logger for mocking
+logger = logging.getLogger("utils.hosts")
```

Comment:
Not a big deal, but you could put this on the top of the file. OR you could put it within the `test_clean_milmove_host_users` function to make it clear that's the only place where it's used. Either way though, doesn't really matter